### PR TITLE
Remove Edge reporting.

### PR DESCRIPTION
**Pull request by bianpengyuan**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by mandarjog — APPROVED** [review]
Lgtm

File: extensions/stackdriver/config/v1alpha1/stackdriver_plugin_config.proto [hunk 1]
```diff
@@ -80,7 +80,7 @@ message PluginConfig {
 
   // Optional. Controls whether or not to export mesh edges to a mesh edges
   // service. This is disabled by default.
-  bool enable_mesh_edges_reporting = 3;
+  bool enable_mesh_edges_reporting = 3 [deprecated = true];
```
Comment (douglas-reid) [inline extensions/stackdriver/config/v1alpha1/stackdriver_plugin_config.proto:83]: should we update the comment here as well. Something to the effect of `Deprecated -- Mesh edge reporting is no longer supported` ?

Comment (bianpengyuan) [reply extensions/stackdriver/config/v1alpha1/stackdriver_plugin_config.proto:84]: Done.

Comment (mandarjog) [reply extensions/stackdriver/config/v1alpha1/stackdriver_plugin_config.proto:84]: Should it be stronger than deprecated,  “This setting has no effect”

Comment (bianpengyuan) [reply extensions/stackdriver/config/v1alpha1/stackdriver_plugin_config.proto:84]: Updated to make it more clear that this option is no-op.

**Comment by bianpengyuan** [comment]
@kyessenov could you please help to approve this? Want to get this in and then follow up with an Envoy update PR.

**Review by kyessenov — APPROVED** [review]
With pleasure!
/approve

**Pull request merged**